advanced/wallets/react-wallet-v2/src/lib/smart-accounts/SmartAccountLib.ts

```diff
@@ -173,6 +173,33 @@ export abstract class SmartAccountLib implements EIP155Wallet {
     if (!this.client || !this.client.account) {
       throw new Error('Client not initialized')
     }
+    const accountDeployed = await isSmartAccountDeployed(this.publicClient,this.client.account.address)
+    /**
+     * this is just a temporary fix for safe7579 modular account
+     * is safe7579Account is not depoyed the need to create 
+     * one useroperation which first deploy and setup the account ,
+     * second user operation will be used to call the desired actions 
+     * */ 
+    if(this.type === "Safe" && this.client.name === "Safe7579SmartAccount" && !accountDeployed){
+      const setUpUserOp = await this.client.prepareUserOperationRequest({
+        userOperation: {
+          callData: await this.client.account.encodeCallData({ to, value, data }),
+        },
+        account: this.client.account,
+        })
+        const newSignature = await this.client.account.signUserOperation(setUpUserOp)
+        console.log('Signatures',{old: setUpUserOp.signature, new: newSignature});
+
+        setUpUserOp.signature = newSignature
+
+        const setUpUserOpHash = await this.bundlerClient.sendUserOperation({
+        userOperation: setUpUserOp
+        })
+        const txHash = await this.bundlerClient.waitForUserOperationReceipt({
+          hash:setUpUserOpHash
+        })
+        console.log(`safe setup txHash: ${txHash}`)
+    }
```

Comment:
Why is this in `SmartAccountLib` instead of `SafeSmartAccountLib`?

Reply:
my fault, I will separate it to SafeSmartAccountLib.